mongoimport/mongoimport.go

```diff
@@ -512,49 +501,25 @@ readLoop:
 	return filterIngestError(imp.IngestOptions.StopOnError, err)
 }
 
-type upserter struct {
-	imp         *MongoImport
-	collection  *mongo.Collection
-	insertOpts  *driverOpts.InsertOneOptions
-	replaceOpts *driverOpts.ReplaceOptions
-	updateOpts  *driverOpts.UpdateOptions
-}
-
-func (imp *MongoImport) newUpserter(collection *mongo.Collection, bypassDocValidation bool) *upserter {
-	return &upserter{
-		imp:         imp,
-		collection:  collection,
-		insertOpts:  driverOpts.InsertOne().SetBypassDocumentValidation(bypassDocValidation),
-		replaceOpts: driverOpts.Replace().SetBypassDocumentValidation(bypassDocValidation).SetUpsert(true),
-		updateOpts:  driverOpts.Update().SetBypassDocumentValidation(bypassDocValidation).SetUpsert(true),
+func (imp *MongoImport) insertDocument(inserter *db.BufferedBulkInserter, document bson.D) error {
```

Comment:
I would suggest renaming this to something more generic since we're doing merges and upserts here too, not just inserts. Maybe `importDocument`?

Reply:
done